### ISSUE TITLE
Blocking I/O considered harmful

### ISSUE THREAD

**Issue by jbearer** [EspressoSystems/seahorse#58]
We mix blocking I/O with async code. We should not do this. An example of why this is bad:

The CLI uses blocking I/O. This causes `test_view_freeze` to hang when run on a system with fewer than 4 cores. The test has one main task, and it spawns 3 tasks to run 3 wallet CLIs. The main task blocks waiting for the wallets to start while the wallets block waiting for input from the main task. Example trace on a system with 2 cores:
```
main task/core0: spawn Task1 to run wallet 1
main task/core0: block waiting for a prompt from Task1
Task1/core1: print a prompt and then block waiting for input from main task
main task/core0: spawn Task2 to run wallet 2
main task/core0: block waiting for a prompt from Task2
```
At this point, both cores are blocked (main task on core0 and Task1 on core1) so Task2 never gets scheduled, and main is never unblocked. This happens because we use blocking I/O, so whenever a task blocks, it does so without returning to the executor to yield the core to another task.

We can probably fix this by wrapping I/O in `spawn_blocking` futures. It may be a good idea to look into using async I/O to begin with.
